src/OpenTelemetry.Exporter.Geneva/Internal/TableNameSerializer.cs

```diff
@@ -0,0 +1,313 @@
+                {
+                    tempTableMappings[kv.Key] = BuildStr8BufferForAsciiString(kv.Value);
+                }
+            }
+
+            this.m_tableMappings = tempTableMappings;
+        }
+    }
+
+    public static bool IsReservedTableName(string tableName)
+    {
+        Debug.Assert(!string.IsNullOrWhiteSpace(tableName), "tableName was null or whitespace");
+
+        // TODO: Implement this if needed.
+
+        return false;
+    }
+
+    public static bool IsValidTableName(string tableName)
+    {
+        Debug.Assert(!string.IsNullOrWhiteSpace(tableName), "tableName was null or whitespace");
+
+        var length = tableName.Length;
+        if (length > MaxSanitizedEventNameLength)
+        {
+            return false;
+        }
+
+        char firstChar = tableName[0];
+        if (firstChar < 'A' || firstChar > 'Z')
+        {
+            return false;
+        }
+
+        for (int i = 1; i < length; i++)
+        {
+            char cur = tableName[i];
+            if ((cur >= 'a' && cur <= 'z') || (cur >= 'A' && cur <= 'Z') || (cur >= '0' && cur <= '9'))
+            {
+                continue;
+            }
+
+            return false;
+        }
+
+        return true;
+    }
+
+    public int ResolveAndSerializeTableNameForCategoryName(byte[] destination, int offset, string categoryName, out ReadOnlySpan<byte> tableName)
+    {
+        byte[] mappedTableName = this.ResolveTableMappingForCategoryName(categoryName);
+
+        if (mappedTableName == s_passthroughTableName)
+        {
+            // Pass-through mode with a full cache.
+
+            int bytesWritten = SanitizedCategoryName(new Span<byte>(destination, offset, MaxSanitizedEventNameLength + 2), categoryName);
+
+            tableName = new ReadOnlySpan<byte>(destination, offset, bytesWritten);
+
+            return offset + bytesWritten;
+        }
+
+        tableName = mappedTableName;
+
+        return MessagePackSerializer.SerializeSpan(destination, offset, tableName);
+    }
+
+    private static byte[] BuildStr8BufferForAsciiString(string value)
+    {
+        var length = value.Length;
+
+        byte[] buffer = new byte[length + 2];
+
+        Encoding.ASCII.GetBytes(value, 0, length, buffer, 2);
+
+        MessagePackSerializer.WriteStr8Header(buffer, 0, length);
+
+        return buffer;
+    }
+
+    // This method would map the logger category to a table name which only contains alphanumeric values with the following additions:
+    // Any character that is not allowed will be removed.
+    // If the resulting string is longer than 50 characters, only the first 50 characters will be taken.
+    // If the first character in the resulting string is a lower-case alphabet, it will be converted to the corresponding upper-case.
+    // If the resulting string still does not comply with Rule, the category name will not be serialized.
+    private static int SanitizedCategoryName(Span<byte> buffer, string categoryName)
```

Comment:
nit: Why remove the prefix `Serialize` from the method name?

Reply:
Never mind! Just noticed that this method does not always write to the "buffer". We could probably call the method `WriteSanitizedCategoryName` 😀

Reply:
I'll rename to `WriteSanitizedCategoryNameToSpan` 🤣